CONTRIBUTING.md

```diff
@@ -61,6 +61,15 @@ our documentation as described in our [support](./SUPPORT.md) page.
   - Add more detailed description in the body of the git message (`git commit -a` to give you more space and time in
     your text editor to write a good message instead of `git commit -am`)
 
+#### Branch protection rules
+
+This repository has the following branch protection rules in place:
+
+- **Pushing branches that contain the "internal/" prefix is not allowed.** This ensures internal development branches are not accidentally or purposefully pushed to this repo.
+- **Two approvers are required for all merges to main and release branches.** This ensures all code that is approved for release to production is appropriately reviewed. This rule applies to all branches with `*release*` in the branch name.
```

Comment:
Info: we should also update https://github.com/nginx/documentation/blob/main/.github/pull_request_template.md#checklist accordingly. I know @ADubhlaoich is currently working on this file.